cpp/src/reductions/scan/rank_scan.cu

```diff
@@ -51,20 +67,33 @@ std::unique_ptr<column> rank_generator(column_view const& order_by,
                                        rmm::cuda_stream_view stream,
                                        rmm::mr::device_memory_resource* mr)
 {
-  auto comp = cudf::experimental::row::equality::self_comparator(table_view{{order_by}}, stream);
-  auto const device_comparator =
-    comp.equal_to(nullate::DYNAMIC{has_nested_nulls(table_view({order_by}))});
+  auto const order_by_view = table_view{{order_by}};
```

Comment:
Nit: Use `_tview` or `_cv`/`_cview` to have a more expressive var name. You can't know if it is a table view or a column view with just `_view`.